src/github.com/getlantern/balancer/dialer.go

```diff
@@ -48,45 +48,49 @@ type Dialer struct {
 }
 
 var (
-	longDuration    = 1000000 * time.Hour
 	maxCheckTimeout = 5 * time.Second
 )
 
+type metrics struct {
+	avgConnTime     int64
+	consecSuccesses int32
+	consecFailures  int32
+}
 type dialer struct {
 	*Dialer
-	active  int32
-	closeCh chan interface{}
-	errCh   chan time.Time
+	consecSuccesses int32
+	consecFailures  int32
+	// It's actually the average of last connect time and previous average. so
+	// if the connect time for i iteration is t[i], after n iteration, its value
+	// will be 1/2(t[n] + 1/2(t[n-1] + 1/2(t[n-2) + ... + t[1]))...), most
+	// recent connect time contributes most to the value, seems a good indicator.
+	avgConnTime int64
+	closeCh     chan struct{}
```

Comment:
I changed it to chan of [empty struct](http://dave.cheney.net/2014/03/25/the-empty-struct) mostly because `chan interface{}` accepts any type while `chan struct{}` can only accept `chan struct{}{}`.